### DISCUSSION TITLE
Using the CH376 as a USB HOST to get data from non storage devices

### DISCUSSION THREAD

**Discussion by ProgramacaoSunnytec** [djuseeq/Ch376msc#84] [Q&A]
Hi to all!

I was wondering if I can use this IC as an USB host to receive data from a Scale that I have, whitch sends the weight trough serial over a USB connection.

Thank you!

**Comment by kbrworks** [comment]
I just found another method to write and read new files using the CH376S.

I purchased old stock without knowing about the issue with the outdated firmware of the CH376S.

However, I found a solution.

When formatting the USB memory on Windows 10 or 11, the process is incorrect, and the files become invisible. Therefore, I formatted it on Windows XP and created a TEXT.TXT file.

This way, the file is visible when read on the CH376S, and the TEST1.TXT file generated in the CH376S example is displayed normally when placed on an UNO and read on another PC.

You should give it a try, too.
